libsolidity/interface/CompilerStack.h

```diff
@@ -342,11 +342,11 @@ class CompilerStack: boost::noncopyable
 		std::string yulIROptimized; ///< Optimized experimental Yul IR code.
 		std::string ewasm; ///< Experimental Ewasm text representation
 		evmasm::LinkerObject ewasmObject; ///< Experimental Ewasm code
-		mutable std::unique_ptr<std::string const> metadata; ///< The metadata json that will be hashed into the chain.
-		mutable std::unique_ptr<Json::Value const> abi;
-		mutable std::unique_ptr<Json::Value const> storageLayout;
-		mutable std::unique_ptr<Json::Value const> userDocumentation;
-		mutable std::unique_ptr<Json::Value const> devDocumentation;
+		mutable std::optional<std::string> metadata; ///< The metadata json that will be hashed into the chain.
```

Comment:
I don't understand - what is the difference between `metadata` and `sourceMapping`?

Reply:
A raw pointer is returned to sourceMapping in CompilerStack.cpp L587 (and similarly for runtimeSourceMapping), so I didn't change it out of an abundance of caution and lifetime concerns.

Reply:
But you can return a pointer to the string inside the optinoal, can't you? Or would that rather be a `optional<string> const&`?

Reply:
Both of those are options. I would love to take the opportunity to get rid of the pointer return, but that looks like it would require changing some other stuff, so I'll do the first thing.